Chess/desk.cpp

```diff
@@ -1,23 +1,120 @@
-    for (int i=0; i<8; i++) {
-        std::vector<QObject*> row;
-        buffer_.push_back(row);     //it will copy row to buffer_ (now mamory leek [?])
+    for (int row=0; row<max_row_cnt_; row++) {
+        std::vector<Figure*> row_vec;
+        for (int col=0; col<max_row_cnt_; col++)
+            row_vec.push_back(NULL);
+
+        buffer_.push_back(row_vec);     //it will copy row to buffer_ (now mamory leek [?])
+    }
+
+
+    if (!defaultStateFilled_) {
+        defaultState_.push_back(CellInfo( Cell(0,0), ROOK,   BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,1), KNIGHT, BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,2), BISHOP, BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,3), QUEEN,  BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,4), KING,   BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,5), BISHOP, BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,6), KNIGHT, BLACK ));
+        defaultState_.push_back(CellInfo( Cell(0,7), ROOK,   BLACK ));
+        for (int i=0; i<8; i++) {
+            defaultState_.push_back(CellInfo( Cell(1,i), PAWN, BLACK ));
+        }
 
-        for (int j=0; j<8; j++) {
-            //row.push_back(new Figure(this));
-            row.push_back(NULL);
+        defaultState_.push_back(CellInfo( Cell(7,0), ROOK,   WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,1), KNIGHT, WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,2), BISHOP, WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,3), QUEEN,  WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,4), KING,   WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,5), BISHOP, WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,6), KNIGHT, WHITE ));
+        defaultState_.push_back(CellInfo( Cell(7,7), ROOK,   WHITE ));
+        for (int i=0; i<8; i++) {
+            defaultState_.push_back(CellInfo( Cell(6,i), PAWN, WHITE ));
         }
+
+        defaultStateFilled_ = true;
     }
 }
 
 void Desk::fillDefault()
 {
+    for (int i=0; i<defaultState_.size(); i++) {
+        const CellInfo item = defaultState_[i];
+
+        const int row = item.cell_.row_;
+        const int col = item.cell_.col_;
+
+        switch (item.ftype_) {
+        case KING:
+            buffer_[row][col] = new King(this, item.fcolor_);
+            break;
+        default:
+            buffer_[row][col] = new Queen(this, item.fcolor_);
+            break;
+        }
+    }
+
+
+//    for (std::vector<CellInfo>::const_iterator it=defaultState_.begin(); it != defaultState_.end(); it++) {
+//        const size_t x = it->cell_.x_;
+//        const size_t y = it->cell_.y_;
+//        switch (it->ftype_) {
+//        case Figure::KING:
+//            buffer_[x][y] = new King(this, it->fcolor_);
+//            break;
+//        default:
+//            buffer_[x][y] = new Queen(this, it->fcolor_);
+//            break;
+//        }
+//    }

```

Comment:
Try not to commit commented code. If you need it - it should be uncommented If not - deleat it